sklearn/feature_extraction/tests/test_text.py

```diff
@@ -390,15 +383,10 @@ def test_tfidf_no_smoothing():
     tr = TfidfTransformer(smooth_idf=False, norm='l2')
 
     clean_warning_registry()
-    with warnings.catch_warnings(record=True) as w:
-        1. / np.array([0.])
-        numpy_provides_div0_warning = len(w) == 1
```

Comment:
This was added for some numpy versions and architectures in 2012 https://github.com/scikit-learn/scikit-learn/pull/928 since CI passes we can probably remove this indeed.

We can remove clean_warning_registry as well.